services/QuillLMS/app/services/stripe_integration/webhooks/subscription_creator.rb

```diff
@@ -24,13 +24,22 @@ def run
         raise NilPurchaserEmailError if purchaser_email.nil?
         raise NilStripePriceIdError if stripe_price_id.nil?
         raise NilStripeInvoiceIdError if stripe_invoice.id.nil?
-        raise DuplicateSubscriptionError if purchaser.subscription.present?
+        raise DuplicateSubscriptionError if duplicate_subscription?
 
         subscription
         save_stripe_customer_id
         run_plan_custom_tasks
       end
 
+      private def duplicate_subscription?
+        case plan
+        when Plan.stripe_teacher_plan
+          purchaser.subscription&.plan == plan
+        when Plan.stripe_school_plan
+          purchaser.subscriptions.active.any? { |subscription| subscription.schools.pluck(:id).sort == school_ids }
```

Comment:
If a user is an admin for a school, but does not belong to that school, I don't think that school's subscription will come up under `purchaser.subscriptions`, will it?

Reply:
Good point.  I will fix that.

Reply:
@emilia-friedberg I'm going to refactor and add tests tomorrow but I'm thinking something like this?

```
        when Plan.stripe_school_plan
          purchaser.subscriptions.active.any? { |subscription| subscription.schools.pluck(:id).sort == school_ids } ||
            purchaser.associated_schools.any? { |school| school.subscriptions.active.any?  { |subscription| subscription.schools.pluck(:id).sort == school_ids } }
        end
```

Reply:
That makes sense to me. I'm not sure you need the first part of the `or` - I think the second case will always cover the first (because if a user can only have a school subscription if they are associated with a school with a subscription).